packages/typespec-client-generator-core/design-docs/mutliple-services.md

```diff
@@ -0,0 +1,174 @@
+
+```python
+client = ComputeManagementClient(credential=DefaultAzureCredential(), subscription_id="{subscription-id}")
+client.disks.list_by_resource_group(resource_group_name="myResourceGroup")
+client.virtual_machine_sizes.list(location="eastus")
+```
+
+## First Step Design
+
+The initial design focuses on the scenario of auto-merging multiple services into a single client. For other scenarios, such as redefining client hierarchy for multiple services, we will consider them in future iterations.
+
+### Syntax Proposal
+
+Previously, the `service` property of the `@client` option only supported a single service. We propose extending it to accept an array of services.
+
+For example, given two services, the original service specs are:
+
+```typespec title="ServiceA/main.tsp"
+@service
+@versioned(VersionsA)
+namespace ServiceA;
+
+enum VersionsA {
+  av1,
+  av2,
+}
+interface AI {
+  @route("/aTest")
+  aTest(@query("api-version") apiVersion: VersionsA): void;
+}
+```
+
+```typespec title="ServiceB/main.tsp"
+@service
+@versioned(VersionsB)
+namespace ServiceB;
+
+enum VersionsB {
+  bv1,
+  bv2,
+}
+interface BI {
+  @route("/bTest")
+  bTest(@query("api-version") apiVersion: VersionsB): void;
+}
+```
+
+To define a combined client:
+
+```typespec title="client.tsp"
+import "./ServiceA/main.tsp";
+import "./ServiceB/main.tsp";
+import "@azure-tools/typespec-client-generator-core";
+
+using Azure.ClientGenerator.Core;
+
+@client({
+  name: "CombineClient",
+  service: [ServiceA, ServiceB],
+})
+@useDependency(ServiceA.VersionsA.av2, ServiceB.VersionsB.bv2) // optional
+namespace CombineClient;
+```
+
+**Explanation:**
+
+- `@client` with the `service` property as an array indicates a combined client.
+- `@useDependency` specifies the version for each service:
+  - If all services are unversioned, `@useDependency` can be omitted.
+  - If any service is versioned and `@useDependency` is not set, TCGC will use the latest version for each service by default.
+- Only one `@client` with multiple services can be defined per package. Defining multiple such clients or mixing with `@operationGroup` is not supported.
+
+### TCGC Behavior
+
+When TCGC detects multiple services in one client, it will:
+
+1. Create the root client for the combined client. If any service is versioned, the root client's initialization method will have an `apiVersion` parameter with no default value. The `apiVersions` property and the `apiVersion` parameter for the root client will be empty (since multiple services' API versions cannot be combined).
+2. Create sub-clients for each service's nested namespaces or interfaces. Each sub-client will have its own `apiVersion` property and initialization method if the service is versioned.
+3. Operations directly under each service's namespace are placed under the root client. Operations under nested namespaces or interfaces are placed under the corresponding sub-clients.
```

Comment:
First time I read this I understood that operations directly under the service namespace are placed in the root/combined client. But I think they will be within the service subclient right?

```ts
namespace ServiceA {
  op createA(): void;
}

namespace ServiceB {
   op createB(): void;
}
```

We get 

```ts
client.serviceA.createA()
client.serviceB.createB()
```

Not 
```ts
client.createA()
client.createB()
```